src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -401,14 +435,34 @@ NVTX.@annotate function set_diagnostic_edmf_precomputed_quantities_do_integral!(
             # Updraft q_tot sources from precipitation formation
             # To be applied in updraft continuity, moisture and energy
             # for updrafts and grid mean
-            @. S_q_totʲ_prev_level = q_tot_precipitation_sources(
-                precip_model,
-                thermo_params,
-                microphys_params,
-                dt,
-                q_totʲ_prev_level,
-                tsʲ_prev_level,
-            )
+            if precip_model isa Microphysics0Moment
+                @. S_q_totʲ_prev_level = q_tot_precipitation_sources(
+                    precip_model,
+                    thermo_params,
+                    microphys_params,
+                    dt,
+                    q_totʲ_prev_level,
+                    tsʲ_prev_level,
+                )
+            end
+            if precip_model isa Microphysics1Moment
```

Comment:
```suggestion
            elseif precip_model isa Microphysics1Moment
```
(It doesn't really matter I think, but reduces one line:)